Frontend/ui-driver/src/Screens/DriverEarningsScreen/ComponentConfig.purs

```diff
@@ -0,0 +1,253 @@
+import Components.PrimaryButton.View as PrimaryButton
+import Components.RequestInfoCard as RequestInfoCard
+import Data.Maybe (isJust)
+import Engineering.Helpers.Utils (getCurrentDay)
+import Font.Style (Style(..))
+import Helpers.Utils as HU
+import Language.Strings (getString)
+import Language.Types (STR(..))
+import Prelude ((<>), (==), (*), show, not)
+import Data.Int (toNumber)
+import PrestoDOM (Gravity(..), Length(..), Margin(..), Padding(..), Visibility(..), background)
+import PrestoDOM.Types.DomAttributes (Corners(..))
+import Screens.Types as ST
+import Styles.Colors as Color
+
+primaryButtonConfig :: Boolean -> PrimaryButtonConfig.Config
+primaryButtonConfig isActive = 
+  PrimaryButtonConfig.config { 
+    textConfig
+      { text = getString BUY_NOW
+      , accessibilityHint = (" : Buy Now")
+      }
+    , cornerRadius = 6.0
+    , margin = Margin 16 24 16 16
+    , isClickable = isActive
+    , alpha = if isActive then 1.0 else 0.5
+    }
+
+genericHeaderConfig :: ST.DriverEarningsScreenState -> GenericHeaderConfig.Config
+genericHeaderConfig state = 
+  let headerText = if state.props.subView == ST.FAQ_VIEW then getString YATRI_COINS_FAQS
+                    else if state.props.subView == ST.FAQ_QUESTON_VIEW then ""
+                    else if not state.data.config.coinsConfig.enableYatriCoins then getString EARNINGS
+                    else getString USE_COINS
+  in
+  GenericHeaderConfig.config{ 
+    height = WRAP_CONTENT
+    , padding = if state.props.subView == ST.EARNINGS_VIEW then Padding 16 16 16 16 else PaddingVertical 5 5
+    , textConfig {
+        text = headerText
+      , color = Color.darkCharcoal }
+    , prefixImageConfig {
+        height = V 25
+      , width = V 25
+      , imageUrl = "ny_ic_chevron_left"
+      , visibility = if state.data.config.coinsConfig.enableYatriCoins then VISIBLE else GONE
+      } 
+    , suffixImageConfig {
+        visibility = GONE }
+    , margin = MarginBottom if state.data.config.coinsConfig.enableYatriCoins then 0 else 24
+  }
+
+earningsPopupConfig :: ST.DriverEarningsScreenState -> PopUpModalConfig.Config
+earningsPopupConfig state =
+  PopUpModalConfig.config {
+      cornerRadius = Corners 15.0 true true true true
+      , margin = MarginHorizontal 16 16
+      , padding = Padding 16 16 16 16
+      , gravity = CENTER
+      , backgroundColor =  Color.black9000
+      , backgroundClickable = true
+      , buttonLayoutMargin = MarginBottom 0
+      , optionButtonOrientation = case state.props.popupType of
+                  ST.COIN_TO_CASH_POPUP -> "HORIZONTAL"
+                  ST.COIN_TO_CASH_FAIL_POPUP -> "VERTICAL"
+                  ST.COINS_EXPIRING_POPUP -> "VERTICAL"
+                  _ -> "HORIZONTAL"
+    ,primaryText {
+        text = case state.props.popupType of
+                  ST.COIN_TO_CASH_POPUP -> show (state.data.coinConversionRate * toNumber state.data.coinsToUse) <> " " <> getString WILL_BE_ADJUSTED_IN_YOUR_FUTURE_SUBSCRIPTION_DUES
```

Comment:
combine all the text which uses this props to update and retur record from a function using this condition.